# HG changeset patch
# User Rik <rik@octave.org>
# Date 1449937882 28800
#      Sat Dec 12 08:31:22 2015 -0800
# Node ID 61892a245653c0a09f5cc4697e591c9d81d2f089
# Parent  1f4d6b1f50c1912dee8b9e46e2c2b730cf0c15e2
2015 Code Sprint: dmperm.cc: use ovl().

diff --git a/libinterp/dldfcn/dmperm.cc b/libinterp/dldfcn/dmperm.cc
--- a/libinterp/dldfcn/dmperm.cc
+++ b/libinterp/dldfcn/dmperm.cc
@@ -107,25 +107,21 @@ dmperm_internal (bool rank, const octave
       CXSPARSE_NAME (d) *dm = CXSPARSE_NAME(_dmperm) (&csm, 0);
 #else
       CXSPARSE_NAME (d) *dm = CXSPARSE_NAME(_dmperm) (&csm);
 #endif
 
       //retval(5) = put_int (dm->rr, 5);
       //retval(4) = put_int (dm->cc, 5);
 #if defined (CS_VER) && (CS_VER >= 2)
-      retval(3) = put_int (dm->s, dm->nb+1);
-      retval(2) = put_int (dm->r, dm->nb+1);
-      retval(1) = put_int (dm->q, nc);
-      retval(0) = put_int (dm->p, nr);
+      retval = ovl (put_int (dm->p, nr), put_int (dm->q, nc),
+                    put_int (dm->r, dm->nb+1), put_int (dm->s, dm->nb+1));
 #else
-      retval(3) = put_int (dm->S, dm->nb+1);
-      retval(2) = put_int (dm->R, dm->nb+1);
-      retval(1) = put_int (dm->Q, nc);
-      retval(0) = put_int (dm->P, nr);
+      retval = ovl (put_int (dm->P, nr), put_int (dm->Q, nc),
+                    put_int (dm->R, dm->nb+1), put_int (dm->S, dm->nb+1));
 #endif
       CXSPARSE_NAME (_dfree) (dm);
     }
 
   return retval;
 }
 #endif
 
